### DISCUSSION TITLE
Tiny usb UAC2+HID audio control?

### DISCUSSION THREAD

**Discussion by cdg66** [hathach/tinyusb#796] [Q&A] [answered]
Hi,
First of all, thank a lot for your work it's just magical.
Second, would it be possible to have a composite device HID keyboard for audio volume control and the headset example for streaming audio to an i2s amp? I'm working on that I just want to know if it's feasible with the current device stack.

**Comment by hathach** [comment, marked as answer]
audio driver is still work in progress, you will need to dive into code stack should there is something missing or not supported. You could try out some of the audio example to see if that works for you.

**Comment by AaronD-GH** [comment]
This thread is a year and a half old now.  Is the audio driver still a work in progress?

https://github.com/hathach/tinyusb/issues/1186#issuecomment-1287874463
https://forums.raspberrypi.com/viewtopic.php?t=341735

I'm also making a USB sound card with I2S output, with some DSP in between, on a Pi Pico (RP2040).  Ideally UAC2 for the audio, and HID for the DSP settings.  Since the DSP settings include and go far beyond the controls that UAC provides, I'll just do everything in the DSP and manage it all via HID, which leaves the UAC controls unused, ignored, or not even present.
(or if it makes things work better to have the UAC volume and mute functions at the front of the DSP chain, I guess I can do that)
